src/material-experimental/mdc-progress-spinner/progress-spinner.ts

```diff
@@ -0,0 +1,268 @@
+/**
+ * @license
+ * Copyright Google LLC All Rights Reserved.
+ *
+ * Use of this source code is governed by an MIT-style license that can be
+ * found in the LICENSE file at https://angular.io/license
+ */
+
+import {
+  AfterViewInit,
+  ChangeDetectionStrategy,
+  Component,
+  ElementRef,
+  Inject,
+  Input,
+  OnDestroy,
+  Optional,
+  ViewEncapsulation
+} from '@angular/core';
+import {
+  MDCCircularProgressAdapter,
+  MDCCircularProgressFoundation
+} from '@material/circular-progress';
+import {CanColor, CanColorCtor, mixinColor} from '@angular/material/core';
+import {ANIMATION_MODULE_TYPE} from '@angular/platform-browser/animations';
+import {
+  MAT_PROGRESS_SPINNER_DEFAULT_OPTIONS,
+  MatProgressSpinnerDefaultOptions
+} from '@angular/material/progress-spinner';
+import {coerceNumberProperty} from '@angular/cdk/coercion';
+
+// Boilerplate for applying mixins to MatProgressBar.
+/** @docs-private */
+class MatProgressSpinnerBase {
+  constructor(public _elementRef: ElementRef) {
+  }
+}
+
+const _MatProgressSpinnerMixinBase: CanColorCtor & typeof MatProgressSpinnerBase =
+  mixinColor(MatProgressSpinnerBase, 'primary');
+
+/** Possible mode for a progress spinner. */
+export type ProgressSpinnerMode = 'determinate' | 'indeterminate';
+
+/**
+ * Base reference size of the spinner.
+ * @docs-private
+ */
+const BASE_SIZE = 100;
+
+/**
+ * Base reference stroke width of the spinner.
+ * @docs-private
+ */
+const BASE_STROKE_WIDTH = 10;
+
+@Component({
+  selector: 'mat-progress-spinner',
+  exportAs: 'matProgressSpinner',
+  host: {
+    'role': 'progressbar',
+    'class': 'mat-mdc-progress-spinner',
+    '[class._mat-animation-noopable]': `_noopAnimations`,
+    '[style.width.px]': 'diameter',
+    '[style.height.px]': 'diameter',
+    '[attr.aria-valuemin]': 'mode === "determinate" ? 0 : null',
+    '[attr.aria-valuemax]': 'mode === "determinate" ? 100 : null',
+    '[attr.aria-valuenow]': 'mode === "determinate" ? value : null',
+    '[attr.mode]': 'mode',
+  },
+  inputs: ['color'],
+  templateUrl: 'progress-spinner.html',
+  styleUrls: ['progress-spinner.css'],
+  changeDetection: ChangeDetectionStrategy.OnPush,
+  encapsulation: ViewEncapsulation.None,
+})
+export class MatProgressSpinner extends _MatProgressSpinnerMixinBase implements AfterViewInit,
+  OnDestroy,
+  CanColor {
+  /** Whether the _mat-animation-noopable class should be applied, disabling animations.  */
+  _noopAnimations: boolean;
+  /** Implements all of the logic of the MDC circular progress. */
+  private _foundation: MDCCircularProgressFoundation | undefined;
+  private _rootElement: HTMLElement;
+  private _determinateCircle: HTMLElement;
+  /** Adapter used by MDC to interact with the DOM. */
+  private _adapter: MDCCircularProgressAdapter = {
```

Comment:
Actually we're in the process of refactoring the adapters to classes (e.g. https://github.com/angular/components/pull/19980) so might as well use that style from the start here

Reply:
added a TODO

Reply:
Can we open an issue with MDC and link to that here?

Reply:
https://github.com/material-components/material-components-web/pull/6256

Reply:
please put that link in the `TODO` comment